xml/admin_ceph_upgrade.xml

```diff
@@ -1397,6 +1397,146 @@ data:
    </step>
   </procedure>
  </sect1>
+
+ <sect1 xml:id="filestore2bluestore">
+  <title>OSD Migration to &bluestore;</title>
+  <para>
+   OSD &bluestore; is a new back end for the OSD daemons. It is the default
+   option since &productname; &productnumber;. Compared to FileStore, which
+   stores objects as files in an XFS file system, &bluestore; can deliver
+   increased performance because it stores objects directly on the underlying
+   block device. &bluestore; also enables other features, such as built-in
+   compression and EC overwrites, that are unavailable with FileStore.
+  </para>
+  <para>
+   Specifically for &bluestore;, an OSD has a 'wal' (Write Ahead Log) device
+   and a 'db' (RocksDB database) device. The RocksDB database holds the
+   metadata for a &bluestore; OSD. These two devices will reside on the same
+   device as an OSD by default, but either can be placed on faster/different
+   media.
+  </para>
+  <para>
+   In SES5, both FileStore and &bluestore; are supported and it is possible
+   for FileStore and &bluestore; OSDs to co-exist in a single cluster. During
+   the SUSE Enterprise Storage upgrade procedure, FileStore OSDs are not
+   automatically converted to &bluestore;. Be aware that the
+   &bluestore;-specific features will not be available on OSDs that have not
+   been migrated to &bluestore;.
+  </para>
+  <para>
+   Before converting to &bluestore;, the OSDs need to be running &productname;
+   &productnumber;. The conversion is a slow process as all data gets
+   re-written twice. Though the migration process can take a long time to
+   complete, there is no cluster outage and all clients can continue accessing
+   the cluster during this period. However, do expect lower performance for
+   the duration of the migration. This is caused by rebalancing and
+   backfilling of cluster data.
+  </para>
+  <para>
+   Use the following procedure to migrate FileStore OSDs to &bluestore;:
+  </para>
+  <tip>
+   <title>Turn Off Safety Measures</title>
+   <para>
+    &salt; commands needed for running the migration are blocked by safety
+    measures. In order to turn these precautions off, run the following
+    command:
+   </para>
+ <screen>
+ &prompt.smaster;salt-run disengage.safety
+ </screen>
```

Comment:
I have been **sternly** told off for newlines after <screen> or before </screen>

Reply:
This is the way @tbazant wanted the screen tags to be indented.

Reply:
OK then. Objection withdrawn and may @cwickert have mercy on our souls.

Reply:
I remember that saying too...ancient times now, myeb true, but:
In single line &lt;screens>'s, it's quite ok to have it inline. But with multi-line ones, it is MUCH easier to edit the inner &lt;screen>'s content if it is like this. And honestly, it does not look bad either.